File: java-apps/kq-common/src/main/java/com/kafkaquest/kq/common/util/models/events/responsedata/GameStartedFromSavedStateEventResponseData.java
package com.kafkaquest.kq.common.util.models.events.responsedata;

import com.kafkaquest.kq.common.util.models.events.other.JigsawPieceCurrentPositionResponse;
import com.kafkaquest.kq.common.util.models.events.other.JigsawPuzzlePieceResponse;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@EqualsAndHashCode(callSuper = false)
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class GameStartedFromSavedStateEventResponseData extends ResponseData {
    private long startFromTime;
    private List<JigsawPuzzlePieceResponse> puzzlePieces;
    private List<JigsawPieceCurrentPositionResponse> addedPieces;
}

File: kq-projection-dotnet/kq-projection/Common/Models/Events/GameStartedFromSavedStateEventResponseData.cs
using KQ_Projection.Common.Models.Jigsaw;
using System.Text.Json.Serialization;

namespace KQ_Projection.Common.Models.Events
{
    public class GameStartedFromSavedStateEventResponseData : ResponseData
    {
        [JsonPropertyName("startFromTime")]
        public long startFromTime { get; set; }

        [JsonPropertyName("puzzlePieces")]
        public List<JigsawPuzzlePieceResponse> puzzlePieces { get; set; }

        [JsonPropertyName("addedPieces")]
        public List<JigsawPieceCurrentPositionResponse> addedPieces { get; set; }
    }
}
